client/components/drop-zone/index.jsx

```diff
@@ -1,81 +1,71 @@
 /**
  * External dependencies
  */
-var ReactDom = require( 'react-dom' ),
-	React = require( 'react' ),
-	createFragment = require( 'react-addons-create-fragment' ),
-	without = require( 'lodash/without' ),
-	includes = require( 'lodash/includes' ),
-	classNames = require( 'classnames' ),
-	noop = require( 'lodash/noop' );
+import ReactDom from 'react-dom';
+import React, { Component, PropTypes } from 'react';
+import createFragment from 'react-addons-create-fragment';
+import without from 'lodash/without';
+import includes from 'lodash/includes';
+import classNames from 'classnames';
+import noop from 'lodash/noop';
+import identity from 'lodash/identity';
 
 /**
  * Internal dependencies
  */
-var RootChild = require( 'components/root-child' );
+import RootChild from 'components/root-child';
+import localize from 'lib/mixins/i18n/localize';
 
-module.exports = React.createClass( {
-	displayName: 'DropZone',
+export class DropZone extends Component {
+	constructor( props ) {
+		super( props );
 
-	propTypes: {
-		onDrop: React.PropTypes.func,
-		onVerifyValidTransfer: React.PropTypes.func,
-		onFilesDrop: React.PropTypes.func,
-		fullScreen: React.PropTypes.bool,
-		icon: React.PropTypes.string
-	},
-
-	getInitialState: function() {
-		return {
+		this.state = {
 			isDraggingOverDocument: false,
 			isDraggingOverElement: false
 		};
-	},
-
-	getDefaultProps: function() {
-		return {
-			onDrop: noop,
-			onVerifyValidTransfer: () => true,
-			onFilesDrop: noop,
-			fullScreen: false,
-			icon: 'dashicons dashicons-admin-media'
-		};
-	},
 
-	componentDidMount: function() {
+		// Bind listeners found in componentDidMount(), except for the ones whose
+		// implementation doesn't refer to `this`.
+		this.onDrop = this.onDrop.bind( this );
```

Comment:
I found https://facebook.github.io/react/docs/reusable-components.html#no-autobinding, which is what I'm following here.

Since this is a refactoring, I wanted to keep method names so I wouldn't have to worry about the component's consumers.

> but I suppose there's rarely a case for needing the original (prefixing does help to _indicate_ that it's a bound function).

Yeah -- in old-style React components, the auto-binding would've overridden the original ones in all practical cases while keeping the original name. Not sure there's much value in keeping the original methods around since even with tests, the `this` bind seems to make only sense -- any practical examples for using the unbound methods you can think of?